### ISSUE TITLE
Atlas: attachment downloader starves the P2P state-machine of event IDs

### ISSUE THREAD

**Issue by jcnelson** [blockstack/stacks-blockchain#2428]
The attachment downloader needs to keep track of dead conversations' events and return them to the p2p state machine, so their associated event-polling state can be freed.  Otherwise, the node will no longer respond to network requests once all event ID slots are taken.

**Comment by lgalabru** [comment]
Should I wait on https://github.com/blockstack/stacks-blockchain/issues/2429#issuecomment-776060683 for taking on this?

**Comment by jcnelson** [comment]
This needs to be addressed before Atlas can be re-enabled.  But, it's not blocked on #2429 -- the block downloader doesn't need #2429 to work, for example.  I think #2429 is its own PR after Atlas, since it's going to cut across several different modules.
